walkthroughs/howto-ingress-enhancements/README.md

```diff
@@ -0,0 +1,437 @@
+          }]
+        },
+        "match": {
+          "headers": [{
+            "name": "color_header",
+            "match": {
+              "prefix": "redoryellow"
+            }
+          }],
+          "prefix": "/",
+          "queryParameters": [{
+            "name": "color",
+            "match": {
+              "exact": "yellow"
+            }
+          }],
+        }
+      }
+    }
+  }       
+    ```
+ 
+
+## Setup
+For the Color App setup, we'll use an NLB to forward traffic to the Virtual Gateway (running a set of Envoys). We would configure 2 Gateway Routes - red and yellow pointing to 2 Virtual Services.
+
+Let's now jump into the example.
+
+## Step 1: Prerequisites
+
+
+1. This walkthrough makes use of the unix command line utility `jq`. If you don't already have it, you can install it from [here](https://stedolan.github.io/jq/).
+
+2. This example uses features in the [App Mesh Preview Channel](https://docs.aws.amazon.com/app-mesh/latest/userguide/preview.html). You'll need to install the latest `appmesh-preview`      model.
+
+   Add the Preview Channel service model to the AWS CLI version 1 with the following command:
+   ```bash
+       aws configure add-model \
+           --service-name appmesh-preview \
+           --service-model https://raw.githubusercontent.com/aws/aws-app-mesh-roadmap/master/appmesh-preview/service-model.json
+   ```
+   
+   If you're using the AWS CLI version 2, add the service model with the following commands:
+   ```bash
+       curl -o service-model.json https://raw.githubusercontent.com/aws/aws-app-mesh-roadmap/master/appmesh-preview/service-model.json
+       aws configure add-model --service-name appmesh-preview --service-model file://service-model.json
+   ```
+
+3. Install Docker. It is needed to build the demo application images.
+
+4. You'll need a keypair stored in AWS to access a bastion host. You can create a keypair using the command below if you don't have one. See [Amazon EC2 Key Pairs](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/ec2-key-pairs.html).
+
+```bash
+aws ec2 create-key-pair --key-name color-app-ingress-v2 | jq -r .KeyMaterial > ~/.ssh/color-app-ingress-v2.pem
+chmod go-r ~/.ssh/color-app-ingress-v2.pem
+```
+
+This command creates an Amazon EC2 Key Pair with name `color-app-ingress-v2` and saves the private key at
+`~/.ssh/color-app-ingress-v2.pem`.
+
+## Step 2: Set Environment Variables
+We need to set a few environment variables before provisioning the
+infrastructure. Please change the value for `AWS_ACCOUNT_ID`, `KEY_PAIR_NAME`, and `ENVOY_IMAGE` below.
+
+```bash
+export AWS_ACCOUNT_ID=<your account id>
+export ENVOY_IMAGE=<get the latest from https://docs.aws.amazon.com/app-mesh/latest/userguide/envoy.html>
+export KEY_PAIR_NAME=<color-app-ingress-v2 or your-keypair-name>
+```
+
+Set the following environment variables specific to the walkthrough:
+
+```bash
+export AWS_DEFAULT_REGION=us-west-2
+export ENVIRONMENT_NAME=AppMeshIngressV2Example
```

Comment:
Missed to note this earlier. Could we remove `V2`/`enhancements` from all places and make it `how-to-match-and-rewrite-at-ingress`

Reply:
No problem, updated.